src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1625,7 +1640,16 @@ private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state,
                 try { }
                 finally
                 {
-                    QueueUserWorkItemCallback tpcallBack = new QueueUserWorkItemCallback(callBack, state, compressStack, ref stackMark);
+                    ExecutionContext context = compressStack && !ExecutionContext.IsFlowSuppressed() ?
+                                                ExecutionContext.Capture(ref stackMark, ExecutionContext.CaptureOptions.IgnoreSyncCtx | ExecutionContext.CaptureOptions.OptimizeDefaultCase) :
+                                                ExecutionContext.Default;
+
+                    QueueUserWorkItemCallback tpcallBack = (context == ExecutionContext.Default) ?
+                                                (state == null ? 
+                                                    new QueueUserWorkItemCallback(callBack) : // default or null context, no state 
+                                                    new QueueUserWorkItemCallbackWithState(callBack, state)) : // default or null context
```

Comment:
Worth doing then; however wouldn't effect the use case I was looking at - which is a callback from a native thread so doesn't have an interesting ExecutionContext anyway? :-/

It looks like I can `.Capture` the execution context in CoreCLR, for example before the call out, but doesn't look I can restore it, for it to be recaputred, and then `.Run` on the threadpool later; not sure what the best approach is.